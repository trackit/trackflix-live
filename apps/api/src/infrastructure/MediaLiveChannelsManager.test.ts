import { mockClient } from 'aws-sdk-client-mock';
import {
  BatchUpdateScheduleCommand,
  CreateChannelCommand,
  CreateInputCommand,
  DeleteChannelCommand,
  DeleteInputCommand,
  MediaLiveClient,
  StartChannelCommand,
  StopChannelCommand,
} from '@aws-sdk/client-medialive';
import { MediaLiveChannelsManager } from './MediaLiveChannelsManager';

describe('MediaLive channels manager', () => {
  const mock = mockClient(MediaLiveClient);

  beforeEach(() => {
    mock.reset();
  });

  describe('createChannel', () => {
    it('should create inputs', async () => {
      const { mediaLiveChannelsManager } = setup();
      const eventId = 'dbb682ee-1dd6-4ec6-a666-03b04ace1f9d';
      const packageChannelId = '456789';
      const source = 's3://trackflix-live-demo-videos/oss117.mp4';
      const inputId = '9876543';
      const liveChannelArn =
        'arn:aws:medialive:us-west-2:000000000000:channel:8626488';
      const liveChannelId = '8626488';
      const onAirStartTime = '2025-02-28T10:29:23.890Z';

      mock.on(CreateInputCommand).resolves({
        Input: {
          Id: inputId,
        },
      });
      mock.on(CreateChannelCommand).resolves({
        Channel: {
          Arn: liveChannelArn,
          Id: liveChannelId,
        },
      });

      await mediaLiveChannelsManager.createChannel({
        eventId,
        packageChannelId,
        source,
        onAirStartTime,
      });

      const commandCalls = mock.commandCalls(CreateInputCommand);
      expect(commandCalls).toHaveLength(2);
      expect(commandCalls[0].args[0].input).toEqual({
        Name: expect.stringMatching(/^TrackflixLiveMLIW-.*/),
        Sources: [
          {
            Url: source,
          },
        ],
        Type: 'MP4_FILE',
      });
      expect(commandCalls[1].args[0].input).toEqual({
        Name: expect.stringMatching(/^TrackflixLiveMLI-.*/),
        Sources: [
          {
            Url: source,
          },
        ],
        Type: 'MP4_FILE',
      });
    });

    it('should create channel', async () => {
      const { mediaLiveChannelsManager } = setup();
      const eventId = 'dbb682ee-1dd6-4ec6-a666-03b04ace1f9d';
      const packageChannelId = '456789';
      const source = 's3://trackflix-live-demo-videos/oss117.mp4';
      const inputId = '9876543';
      const liveChannelArn =
        'arn:aws:medialive:us-west-2:000000000000:channel:8626488';
      const liveChannelId = '8626488';
      const onAirStartTime = '2025-02-28T10:29:23.890Z';

      mock.on(CreateInputCommand).resolves({
        Input: {
          Id: inputId,
        },
      });
      mock.on(CreateChannelCommand).resolves({
        Channel: {
          Arn: liveChannelArn,
          Id: liveChannelId,
        },
      });

      await mediaLiveChannelsManager.createChannel({
        eventId,
        packageChannelId,
        source,
        onAirStartTime,
      });

      const commandCalls = mock.commandCalls(CreateChannelCommand);
      expect(commandCalls).toHaveLength(1);
      expect(commandCalls[0].args[0].input).toMatchSnapshot();
    });

    it('should create schedule', async () => {
      const { mediaLiveChannelsManager } = setup();
      const eventId = 'dbb682ee-1dd6-4ec6-a666-03b04ace1f9d';
      const packageChannelId = '456789';
      const source = 's3://trackflix-live-demo-videos/oss117.mp4';
      const inputId = '9876543';
      const liveChannelArn =
        'arn:aws:medialive:us-west-2:000000000000:channel:8626488';
      const liveChannelId = '8626488';
      const onAirStartTime = '2025-02-28T10:29:23.890Z';

      mock.on(CreateInputCommand).resolves({
        Input: {
          Id: inputId,
        },
      });
      mock.on(CreateChannelCommand).resolves({
        Channel: {
          Arn: liveChannelArn,
          Id: liveChannelId,
        },
      });

      await mediaLiveChannelsManager.createChannel({
        eventId,
        packageChannelId,
        source,
        onAirStartTime,
      });

      const commandCalls = mock.commandCalls(BatchUpdateScheduleCommand);
      expect(commandCalls).toHaveLength(1);
      expect(commandCalls[0].args[0].input).toMatchInlineSnapshot(
        {
          ChannelId: liveChannelId,
          Creates: {
            ScheduleActions: [
              {
                ActionName: 'StartContent',
                ScheduleActionSettings: {
                  InputSwitchSettings: {
                    InputAttachmentNameReference: `TrackflixLiveMLIW-${eventId}`,
                  },
                },
                ScheduleActionStartSettings: {
                  FixedModeScheduleActionStartSettings: {
                    Time: onAirStartTime,
                  },
                },
              },
            ],
          },
        },
        `
        {
          "ChannelId": "8626488",
          "Creates": {
            "ScheduleActions": [
              {
                "ActionName": "StartContent",
                "ScheduleActionSettings": {
                  "InputSwitchSettings": {
                    "InputAttachmentNameReference": "TrackflixLiveMLIW-dbb682ee-1dd6-4ec6-a666-03b04ace1f9d",
                  },
                },
                "ScheduleActionStartSettings": {
                  "FixedModeScheduleActionStartSettings": {
                    "Time": "2025-02-28T10:29:23.890Z",
                  },
                },
              },
            ],
          },
        }
      `
      );
    });
  });

  describe('startChannel', () => {
    it('should start channel', async () => {
      const { mediaLiveChannelsManager } = setup();
      const channelId = '812345';

      await mediaLiveChannelsManager.startChannel(channelId);

      const commandCalls = mock.commandCalls(StartChannelCommand);
      expect(commandCalls).toHaveLength(1);
      expect(commandCalls[0].args[0].input).toEqual({
        ChannelId: channelId,
      });
    });
  });

  describe('stopChannel', () => {
    it('should stop channel', async () => {
      const { mediaLiveChannelsManager } = setup();
      const channelId = '812345';

      await mediaLiveChannelsManager.stopChannel(channelId);

      const commandCalls = mock.commandCalls(StopChannelCommand);
      expect(commandCalls).toHaveLength(1);
      expect(commandCalls[0].args[0].input).toEqual({
        ChannelId: channelId,
      });
    });
  });

  describe('deleteChannel', () => {
    it('should delete channel', async () => {
      const { mediaLiveChannelsManager } = setup();
      const channelId = '812345';

      await mediaLiveChannelsManager.deleteChannel(channelId);

      const commandCalls = mock.commandCalls(DeleteChannelCommand);
      expect(commandCalls).toHaveLength(1);
      expect(commandCalls[0].args[0].input).toEqual({
        ChannelId: channelId,
      });
    });
  });

  describe('deleteInput', () => {
    it('should delete input', async () => {
      const { mediaLiveChannelsManager } = setup();
      const inputId = '812345';

      await mediaLiveChannelsManager.deleteInput(inputId);

      const commandCalls = mock.commandCalls(DeleteInputCommand);
      expect(commandCalls).toHaveLength(1);
      expect(commandCalls[0].args[0].input).toEqual({
        InputId: inputId,
      });
    });
  });
});

const setup = () => {
  const client = new MediaLiveClient({
    credentials: {
      accessKeyId: 'fakeAccessKeyId',
      secretAccessKey: 'fakeSecretAccessKey',
    },
  });
  const mediaLiveRoleArn =
    'arn:aws:iam::000000000000:role/trackflix-live-test-MediaLiveRole-8FSnLc3KkA4t';
  const mediaLiveChannelsManager = new MediaLiveChannelsManager({
    client,
    mediaLiveRoleArn,
  });

  return { client, mediaLiveRoleArn, mediaLiveChannelsManager };
};
