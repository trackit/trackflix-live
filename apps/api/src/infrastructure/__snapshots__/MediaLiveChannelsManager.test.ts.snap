// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaLive channels manager createChannel should create channel 1`] = `
{
  "ChannelClass": "SINGLE_PIPELINE",
  "Destinations": [
    {
      "Id": "MainDestination",
      "MediaPackageSettings": [
        {
          "ChannelId": "456789",
        },
      ],
    },
  ],
  "EncoderSettings": {
    "AudioDescriptions": [
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 64000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_1_aac64",
      },
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 64000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_2_aac64",
      },
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 64000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_3_aac64",
      },
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 96000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_1_aac96",
      },
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 96000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_2_aac96",
      },
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 96000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_3_aac96",
      },
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 128000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_1_aac128",
      },
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 128000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_2_aac128",
      },
      {
        "AudioSelectorName": "default",
        "AudioTypeControl": "FOLLOW_INPUT",
        "CodecSettings": {
          "AacSettings": {
            "Bitrate": 128000,
            "RawFormat": "NONE",
            "Spec": "MPEG4",
          },
        },
        "LanguageCodeControl": "FOLLOW_INPUT",
        "Name": "audio_3_aac128",
      },
    ],
    "CaptionDescriptions": [],
    "OutputGroups": [
      {
        "OutputGroupSettings": {
          "MediaPackageGroupSettings": {
            "Destination": {
              "DestinationRefId": "MainDestination",
            },
          },
        },
        "Outputs": [
          {
            "AudioDescriptionNames": [
              "audio_2_aac96",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "960_540",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_960_540",
          },
          {
            "AudioDescriptionNames": [
              "audio_3_aac96",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "1280_720_1",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_1280_720_1",
          },
          {
            "AudioDescriptionNames": [
              "audio_1_aac128",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "1280_720_2",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_1280_720_2",
          },
          {
            "AudioDescriptionNames": [
              "audio_2_aac128",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "1280_720_3",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_1280_720_3",
          },
          {
            "AudioDescriptionNames": [
              "audio_3_aac128",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "1920_1080",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_1920_1080",
          },
          {
            "AudioDescriptionNames": [
              "audio_1_aac64",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "416_234",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_416_234",
          },
          {
            "AudioDescriptionNames": [
              "audio_2_aac64",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "480_270",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_480_270",
          },
          {
            "AudioDescriptionNames": [
              "audio_3_aac64",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "640_360",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_640_360",
          },
          {
            "AudioDescriptionNames": [
              "audio_1_aac96",
            ],
            "CaptionDescriptionNames": [],
            "OutputName": "768_432",
            "OutputSettings": {
              "MediaPackageOutputSettings": {},
            },
            "VideoDescriptionName": "video_768_432",
          },
        ],
      },
    ],
    "TimecodeConfig": {
      "Source": "SYSTEMCLOCK",
    },
    "VideoDescriptions": [
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 200000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CAVLC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 15000,
            "GopBReference": "DISABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 0,
            "GopSize": 30,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_3",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "BASELINE",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 236,
        "Name": "video_416_234",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 416,
      },
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 400000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CAVLC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 15000,
            "GopBReference": "DISABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 0,
            "GopSize": 30,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_3",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "BASELINE",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 272,
        "Name": "video_480_270",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 480,
      },
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 800000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CABAC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 30000,
            "GopBReference": "ENABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 3,
            "GopSize": 60,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_3",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "MAIN",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 360,
        "Name": "video_640_360",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 640,
      },
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 1200000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CABAC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 30000,
            "GopBReference": "ENABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 3,
            "GopSize": 60,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_4_1",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "MAIN",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 432,
        "Name": "video_768_432",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 768,
      },
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 2200000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CABAC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 30000,
            "GopBReference": "ENABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 3,
            "GopSize": 60,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_4_1",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "HIGH",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 540,
        "Name": "video_960_540",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 960,
      },
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 3300000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CABAC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 30000,
            "GopBReference": "ENABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 3,
            "GopSize": 60,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_4_1",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "HIGH",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 720,
        "Name": "video_1280_720_1",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 1280,
      },
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 4700000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CABAC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 30000,
            "GopBReference": "ENABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 3,
            "GopSize": 60,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_4_1",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "HIGH",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 720,
        "Name": "video_1280_720_2",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 1280,
      },
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 6200000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CABAC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 30000,
            "GopBReference": "ENABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 3,
            "GopSize": 60,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_4_1",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "HIGH",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 720,
        "Name": "video_1280_720_3",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 1280,
      },
      {
        "CodecSettings": {
          "H264Settings": {
            "AdaptiveQuantization": "HIGH",
            "AfdSignaling": "NONE",
            "Bitrate": 8000000,
            "ColorMetadata": "INSERT",
            "EntropyEncoding": "CABAC",
            "FlickerAq": "ENABLED",
            "ForceFieldPictures": "DISABLED",
            "FramerateControl": "SPECIFIED",
            "FramerateDenominator": 1001,
            "FramerateNumerator": 30000,
            "GopBReference": "DISABLED",
            "GopClosedCadence": 1,
            "GopNumBFrames": 1,
            "GopSize": 60,
            "GopSizeUnits": "FRAMES",
            "Level": "H264_LEVEL_4_1",
            "LookAheadRateControl": "HIGH",
            "NumRefFrames": 1,
            "ParControl": "SPECIFIED",
            "Profile": "HIGH",
            "RateControlMode": "CBR",
            "ScanType": "PROGRESSIVE",
            "SceneChangeDetect": "ENABLED",
            "SpatialAq": "ENABLED",
            "SubgopLength": "FIXED",
            "Syntax": "DEFAULT",
            "TemporalAq": "ENABLED",
            "TimecodeInsertion": "DISABLED",
          },
        },
        "Height": 1080,
        "Name": "video_1920_1080",
        "RespondToAfd": "NONE",
        "ScalingBehavior": "DEFAULT",
        "Sharpness": 50,
        "Width": 1920,
      },
    ],
  },
  "InputAttachments": [
    {
      "InputAttachmentName": "TrackflixLiveMLIW-dbb682ee-1dd6-4ec6-a666-03b04ace1f9d",
      "InputId": "9876543",
    },
    {
      "InputAttachmentName": "TrackflixLiveMLI-dbb682ee-1dd6-4ec6-a666-03b04ace1f9d",
      "InputId": "9876543",
      "InputSettings": undefined,
    },
  ],
  "InputSpecification": {
    "Codec": "AVC",
    "MaximumBitrate": "MAX_10_MBPS",
    "Resolution": "HD",
  },
  "Name": "TrackflixLiveMLC-dbb682ee-1dd6-4ec6-a666-03b04ace1f9d",
  "RoleArn": "arn:aws:iam::000000000000:role/trackflix-live-test-MediaLiveRole-8FSnLc3KkA4t",
}
`;
